Use autoconf's endian check instead of the locally defined check
so that we can determine the endianess without having to compile
a program.

Idea from http://www.security-lists.org/lists/libnet/2003/03/00001.html

Also remove the old test from aclocal.m4.

Rebased for libnet-1.1.6

diff --git a/configure.in b/configure.in
index e54c688..e2e768e 100644
--- a/configure.in
+++ b/configure.in
@@ -4,6 +4,7 @@ dnl All rights reserved.
 dnl
 dnl Process this file with autoconf to produce a configure script.
 
+AC_PREREQ(2.52)
 AC_INIT(libnet, 1.1.6)
 AC_MSG_RESULT(beginning autoconfiguration process for libnet-${PACKAGE_VERSION} ...)
 AC_CANONICAL_SYSTEM
@@ -38,7 +39,16 @@ dnl
 dnl And some custom things
 dnl
 
-AC_LIBNET_ENDIAN_CHECK
+AC_C_BIGENDIAN([
+ AC_DEFINE(LIBNET_BIG_ENDIAN)
+ ENDIANESS="LIBNET_BIG_ENDIAN"
+ LIBNET_CONFIG_DEFINES="$LIBNET_CONFIG_DEFINES -DLIBNET_BIG_ENDIAN"
+],[
+ AC_DEFINE(LIBNET_LIL_ENDIAN)
+ ENDIANESS="LIBNET_LIL_ENDIAN"
+ LIBNET_CONFIG_DEFINES="$LIBNET_CONFIG_DEFINES -DLIBNET_LIL_ENDIAN"
+], AC_MSG_WARN(cannot detect endianess. try setting ac_cv_c_bigendian to yes or no)) 
+
 dnl AC_LBL_LIBRARY_NET
 
 AC_SUBST(ENDIANESS)
